server/src/main/java/org/apache/druid/server/audit/SQLAuditManager.java

```diff
@@ -268,10 +263,27 @@ private List<AuditEntry> fetchAuditHistoryLastEntries(final String key, final St
           return query
               .bind("type", type)
               .setMaxRows(theLimit)
-              .map((index, r, ctx) -> JacksonUtils.readValue(jsonMapper, r.getBytes("payload"), AuditEntry.class))
+              .map(resultMapper)
               .list();
         }
     );
   }
 
+  private class AuditEventMapper implements ResultSetMapper<AuditEvent>
+  {
+    @Override
+    public AuditEvent map(int index, ResultSet r, StatementContext ctx) throws SQLException
+    {
+      // Read the record and convert to an AuditEvent that can deserialize the payload on-demand
+      AuditRecord record = JacksonUtils.readValue(jsonMapper, r.getBytes("payload"), AuditRecord.class);
```

Comment:
does this not fail when payload is truncated?

Reply:
No, it doesn't fail.
Just to clarify, `payload` in this statement refers to the entire `AuditEntry` as opposed to the payload contained inside the `AuditEntry`.